### PR TITLE
Add ILua.PushGlobalTable

### LINKED ISSUE #81: [ENH] Add a shortcut for pushing Lua Global Table to ILua interface
# Description
Pushing Lua Global table to the stack requires a lot of boilerplate code like `lua.PushSpecial(SPECIAL_TABLES.SPECIAL_GLOB)`. Since working with Global table is the most common operation in GmodDotNet modules, and use cases for other special Lua tables are rare, we should introduce a shortcut for pushing Global table for better developer experience and code readability.

# Possible solution
Add `ILua.PushGlobalTable()` to `GmodNET.API`.

### PR DESCRIPTION
**Pull request by GlebChili**
Closes #81

### CODE REVIEW

File: gm_dotnet_managed/GmodNET.API/ILua.cs [hunk 1]
```diff
@@ -856,6 +856,17 @@ public interface ILua
...
+        /// <summary>
+        /// Pushes Lua Global Table (<see cref="SPECIAL_TABLES.SPECIAL_GLOB"/>) onto the stack.
+        /// </summary>
+        /// <remarks>
+        /// This method is a shortcut for <see cref="ILua.PushSpecial(SPECIAL_TABLES)"/> and <see cref="SPECIAL_TABLES.SPECIAL_GLOB"/> parameter.
+        /// 
+        /// See “Using the Global Table” for more information on Lua Global Table: https://www.lua.org/pil/15.4.html
+        /// </remarks>
+        /// <seealso cref="ILua.PushSpecial(SPECIAL_TABLES)"/>
+        public void PushGlobalTable();
```
Comment (SupinePandora43) [inline gm_dotnet_managed/GmodNET.API/ILua.cs:869]: C#8 invented default interface methods.
you can provide realization of `PushGlobalTable` in interface

**Review by SupinePandora43 — CHANGES_REQUESTED** [review]

File: gm_dotnet_managed/GmodNET.API/ILua.cs (hunk 1 above)
Comment (GlebChili) [reply gm_dotnet_managed/GmodNET.API/ILua.cs:869]: It is pointless in this context, we have only one implementation of ILua interface, so we don't need to worry about extension of contract.

File: gm_dotnet_managed/GmodNET/Lua.cs [hunk 2]
```diff
@@ -647,5 +647,10 @@ public GCHandle PushManagedClosure(Func<ILua, int> function, byte number_of_upva
 
             return delegate_handle;
         }
+
+        public void PushGlobalTable()
```
Comment (SupinePandora43) [inline gm_dotnet_managed/GmodNET/Lua.cs:651]: Can it be turned to lamda-expression?

**Review by SupinePandora43 — CHANGES_REQUESTED** [review]

File: gm_dotnet_managed/GmodNET/Lua.cs (hunk 2 above)
Comment (GlebChili) [reply gm_dotnet_managed/GmodNET/Lua.cs:651]: Do you mean arrow definition?

Comment (SupinePandora43) [reply gm_dotnet_managed/GmodNET/Lua.cs:651]: Exactly

Comment (GlebChili) [reply gm_dotnet_managed/GmodNET/Lua.cs:651]: It's against our code style.

**Pull request merged**